tests/integration/test_charm.py

```diff
@@ -5,19 +5,104 @@
 import logging
 
 import pytest
+from helpers import (
+    check_prometheus_is_ready,
+    get_job_config_for,
+    get_prometheus_config,
+    get_prometheus_rules,
+    get_rules_for,
+)
 
 logger = logging.getLogger(__name__)
 
 
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy_with_ubuntu_image(ops_test):
+@pytest.mark.parametrize("oci_image", ["ubuntu/prometheus:latest", "prom/prometheus:latest"])
+async def test_build_and_deploy_with_alternative_images(ops_test, prometheus_charm, oci_image):
     """Test that the Prometheus charm can be deployed successfully."""
-    prometheus_charm = await ops_test.build_charm(".")
-    resources = {"prometheus-image": "ubuntu/prometheus:latest"}
+    resources = {"prometheus-image": oci_image}
+    app_name = "prometheus-" + oci_image.split("/")[0]
 
+    await ops_test.model.deploy(prometheus_charm, resources=resources, application_name=app_name)
+    await ops_test.model.wait_for_idle(apps=[app_name], status="active")
+    await ops_test.model.block_until(lambda: len(ops_test.model.applications[app_name].units) > 0)
+
+    assert ops_test.model.applications[app_name].units[0].workload_status == "active"
+
+    await check_prometheus_is_ready(ops_test, app_name, 0)
+
+    await ops_test.model.applications[app_name].remove()
+    await ops_test.model.reset()
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy_prometheus_tester(ops_test, prometheus_tester_charm):
+    """Test that Prometheus tester charm can be deployed successfully."""
+    resources = {"prometheus-tester-image": "python:slim"}
+    app_name = "prometheus-tester"
+
+    await ops_test.model.deploy(
+        prometheus_tester_charm, resources=resources, application_name=app_name
+    )
+    await ops_test.model.wait_for_idle(apps=[app_name], status="active")
+    await ops_test.model.block_until(lambda: len(ops_test.model.applications[app_name].units) > 0)
```

Comment:
As above, unless these methods are misleading me, this seems like an odd ordering?

Reply:
I do find that `wait_for_idle` does not wait for units to come up. The doc string of the method only states that it waits for "applications in the model to settle into an idle state" with a timeout. It does not guarantee that the requested number of units will be active.

Reply:
I do find that reversing the ordering makes the test flaky.

Reply:
Will be addressed by https://github.com/canonical/prometheus-operator/issues/154